components/director/internal/domain/runtime_context/repository.go

```diff
@@ -304,6 +304,42 @@ func (r *pgRepository) ListByScenariosAndRuntimeIDs(ctx context.Context, tenant
 	return r.multipleFromEntities(entities), nil
 }
 
+// ListByScenarios lists all runtime contexts that are in any of the given scenarios
+func (r *pgRepository) ListByScenarios(ctx context.Context, tenant string, scenarios []string) ([]*model.RuntimeContext, error) {
+	if len(scenarios) == 0 {
+		return nil, nil
+	}
+	tenantUUID, err := uuid.Parse(tenant)
+	if err != nil {
+		return nil, apperrors.NewInvalidDataError("tenantID is not UUID")
+	}
+
+	var entities RuntimeContextCollection
+
+	// Scenarios query part
+	scenariosFilters := make([]*labelfilter.LabelFilter, 0, len(scenarios))
+	for _, scenarioValue := range scenarios {
+		query := fmt.Sprintf(`$[*] ? (@ == "%s")`, scenarioValue)
```

Comment:
We can extract this common query that is being used in a couple of places. It's a suggestion, not mandatory.